script/gen_requirements_all.py

```diff
@@ -45,20 +47,23 @@ def comment_requirement(req):
     return any(ign in req for ign in COMMENT_REQUIREMENTS)
 
 
-def main():
-    if not os.path.isfile('requirements_all.txt'):
-        print('Run this from HA root dir')
-        return
-
+def gather_modules():
+    """ Collect the information and construct the output. """
     reqs = OrderedDict()
 
     errors = []
+    output = []
+
     for package in sorted(explore_module('homeassistant.components', True)):
         try:
             module = importlib.import_module(package)
         except ImportError:
             errors.append(package)
             continue
+        # For catching the error by RPi.GPIO
+        # RuntimeError: This module can only be run on a Raspberry Pi!
+        except RuntimeError:
```

Comment:
I don't have it installed because it has been commented out in
requirements_all.txt for a while now.

We should however refactor that component to only import it inside the
setup platform method. Loading a file should never cause an error because
it means that the component will not work the first time it gets loaded.

On Sun, Nov 29, 2015, 01:35 Fabian Affolter <notifications@github.com>
wrote:

> In script/gen_requirements_all.py
> <https://github.com/balloob/home-assistant/pull/651#discussion_r46089558>:
>
> >      for package in sorted(explore_module('homeassistant.components', True)):
> >          try:
> >              module = importlib.import_module(package)
> >          except ImportError:
> >              errors.append(package)
> >              continue
> > +        # For catching the error by RPi.GPIO
> > +        # RuntimeError: This module can only be run on a Raspberry Pi!
> > +        except RuntimeError:
>
> Without catching that error I'm not able to use the script. What's the
> trick that you are able to use the script?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/651/files#r46089558>.
>